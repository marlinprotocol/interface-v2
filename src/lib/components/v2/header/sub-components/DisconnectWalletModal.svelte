<script lang="ts">
	import Button from '$lib/atoms/v2/buttons/Button.svelte';
	import Modal from '$lib/atoms/v2/modals/Modal.svelte';
	import TooltipIcon from '$lib/atoms/v2/tooltips/TooltipIcon.svelte';
	import { staticImages } from '$lib/components/images/staticImages';
	import AddTokenPrompt from '$lib/components/v2/prompts/AddTokenPrompt.svelte';
	import { chainConfigStore } from '$lib/data-stores/chainProviderStore';
	import { walletStore } from '$lib/data-stores/walletProviderStore';
	import { handleCopyClick } from '$lib/utils/v2/helpers/componentHelper';

	export let modalFor: string;
	export let disconnect: () => void;
</script>

<Modal {modalFor}>
	<svelte:fragment slot="title">Your wallet</svelte:fragment>
	<!-- <svelte:fragment slot="successmsg">Conversion Successful</svelte:fragment> -->
	<svelte:fragment slot="content">
		<div class="mb-4 rounded-lg bg-base-200 p-4 text-left">
			<div class="flex items-center text-sm font-normal">
				My Address
				<TooltipIcon
					tooltipDirection="tooltip-right"
					tooltipText="Connected wallet address"
					styleClass="ml-1"
				/>
			</div>
			<div class="flex items-center">
				<div
					data-testid="wallet-address"
					class="mt-1 w-[98%] overflow-hidden overflow-ellipsis text-xl font-semibold"
				>
					{$walletStore.address}
				</div>
				<button
					class="shrink-0 cursor-pointer"
					on:click={() => handleCopyClick($walletStore.address, 'Address copied to clipboard')}
				>
					<img width={18} src={staticImages.copyIcon} alt="Copy" />
				</button>
			</div>
		</div>
		<div class="flex flex-wrap gap-4">
			<AddTokenPrompt tokenFor="POND" label="POND" />
			<AddTokenPrompt tokenFor="MPOND" label="MPond" />
			<a
				href="{$chainConfigStore.block_explorer_url}/address/{$walletStore.address}"
				target="_blank"
				rel="noopener noreferrer"
			>
				<Button variant="greyFilled" size="small" styleClass="font-medium gap-3  flex-nowrap">
					<img src={staticImages.OpenInNew} alt="Open" />
					{$chainConfigStore.block_explorer_name}
				</Button>
			</a>
		</div>
	</svelte:fragment>
	<svelte:fragment slot="actionButtons">
		<Button
			variant="filled"
			size="large"
			onclick={disconnect}
			styleClass="w-full font-semibold text-base"
		>
			LOGOUT
		</Button>
	</svelte:fragment>

	<!-- <svelte:fragment slot="errorMessage">
		Insufficient MPOND
	</svelte:fragment> -->
</Modal>
