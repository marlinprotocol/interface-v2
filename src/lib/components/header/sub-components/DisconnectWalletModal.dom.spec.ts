import { render, fireEvent } from '@testing-library/svelte';
import { tick } from 'svelte';
import DisconnectWalletModal from './DisconnectWalletModal.svelte';
const { chainConfigStore, walletStore, chainStore, web3WalletStore } = await vi.hoisted(
	() => import('$lib/data-stores/mock-data-stores/mockStores')
);

beforeAll(() => {
	walletStore.mockSetSubscribeValue({ provider: undefined, signer: undefined, address: '0x123' });
	vi.mock('$lib/data-stores/walletProviderStore', async () => {
		return {
			web3WalletStore,
			walletStore
		};
	});
	vi.mock('$lib/data-stores/chainProviderStore', async () => {
		return {
			chainConfigStore,
			chainStore
		};
	});
});
describe('DisconnectWalletModal', () => {
	const mockFn = vi.fn();
	it('renders properly', () => {
		const { container } = render(DisconnectWalletModal, {
			props: {
				modalFor: 'id-testing',
				disconnect: () => {}
			}
		});
		expect(container).toMatchSnapshot();
	});
	it('calls disconnect function on logout', async () => {
		const { getByText } = render(DisconnectWalletModal, {
			props: {
				modalFor: 'id-testing',
				disconnect: mockFn
			}
		});
		const button = getByText('LOGOUT');
		await fireEvent.click(button);
		await tick(); // Wait for any store updates if needed
		expect(mockFn).toHaveBeenCalledOnce();
	});
	it('has the correct block explorer', () => {
		const { getByText } = render(DisconnectWalletModal, { modalFor: 'wallet', disconnect: mockFn });
		const viewLink = getByText('Block Explorer').closest('a');
		expect(viewLink?.getAttribute('href')).toBe('https://blockexplorer.com/address/0x123');
	});

	it('has the button to add Pond to the users wallet', () => {
		const { getByText } = render(DisconnectWalletModal, { modalFor: 'wallet', disconnect: mockFn });
		expect(getByText('Add POND')).toBeTruthy();
	});

	it('has the button to add MPond to the users wallet', () => {
		const { getByText } = render(DisconnectWalletModal, { modalFor: 'wallet', disconnect: mockFn });
		expect(getByText('Add MPond')).toBeTruthy();
	});

	it('displays the wallet address from the store', async () => {
		const { getByText } = render(DisconnectWalletModal, { modalFor: 'wallet', disconnect: mockFn });
		expect(getByText('0x123')).toBeTruthy();
	});
});
