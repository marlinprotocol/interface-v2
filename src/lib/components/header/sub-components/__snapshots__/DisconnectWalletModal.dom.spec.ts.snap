// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DisconnectWalletModal > renders properly 1`] = `
<body>
  <div>
    <input
      class="modal-toggle s-GZwqZrPaBy6h"
      id="id-testing"
      tabindex="-1"
      type="checkbox"
    />
     
    <div
      class="modal s-GZwqZrPaBy6h"
      data-testid="modal"
    >
      <div
        class="modal-box flex flex-col rounded-3xl bg-white-100 p-0 drop-shadow-sm w-11/12 sm:w-3/4 sm:max-w-[607px] overflow-x-hidden overflow-y-visible s-GZwqZrPaBy6h"
      >
        <div
          class="modal-header sticky top-0 z-10 flex items-center justify-between bg-white-100 px-6 pt-6 s-GZwqZrPaBy6h"
        >
          <div
            class="flex w-full flex-col s-GZwqZrPaBy6h"
          >
             
             
             
            <div
              class="text-left text-[22px] font-medium tracking-[-1px] s-GZwqZrPaBy6h"
            >
              Your wallet
            </div>
             
          </div>
           
          <button
            class="flex h-[54px] w-[54px] shrink-0 cursor-pointer items-center justify-center self-baseline rounded-full border border-grey-100 bg-secondary-content text-sm font-bold text-gray-300 s-GZwqZrPaBy6h"
            data-testid="modal-close-button"
          >
            <img
              alt="Close"
              class="icon-invert s-GZwqZrPaBy6h"
              src="/images/closeIcon.svg"
            />
          </button>
        </div>
         
        <div
          class="modal-body px-6 pt-4 s-GZwqZrPaBy6h"
        >
          <div
            class="mb-4 rounded-lg bg-base-200 p-4 text-left"
          >
            <div
              class="flex items-center text-sm font-normal"
            >
              My Address
				
              <div
                class="tooltip ml-[14px] flex shrink-0 text-left font-normal ml-1 tooltip-secondary tooltip-right"
                data-testid="tooltip-icon"
                data-tip="Connected wallet address"
              >
                <img
                  alt="Info"
                  src="/images/alertV2Icon.svg"
                  width="14px"
                />
              </div>
              <!--&lt;TooltipIcon&gt;-->
            </div>
             
            <div
              class="flex items-center"
            >
              <div
                class="mt-1 w-[98%] overflow-hidden overflow-ellipsis text-xl font-semibold"
                data-testid="wallet-address"
              >
                0x123
              </div>
               
              <button
                class="shrink-0 cursor-pointer"
              >
                <img
                  alt="Copy"
                  src="/images/copyIconNew.svg"
                  width="18"
                />
              </button>
            </div>
          </div>
           
          <div
            class="flex flex-wrap gap-4"
          >
            <button
              class="h-10 btn btn-theme btn-sm items-center flex justify-center disabled:cursor-not-allowed !rounded-[100px] bg-base-200 text-primary border-none hover:bg-base-300 font-medium gap-3"
              id=""
            >
               
              <span
                class="text-2xl font-light"
              >
                +
              </span>
               Add 
              POND
            </button>
            
            <!--&lt;Button&gt;-->
            <!--&lt;AddTokenPrompt&gt;-->
             
            <button
              class="h-10 btn btn-theme btn-sm items-center flex justify-center disabled:cursor-not-allowed !rounded-[100px] bg-base-200 text-primary border-none hover:bg-base-300 font-medium gap-3"
              id=""
            >
               
              <span
                class="text-2xl font-light"
              >
                +
              </span>
               Add 
              MPond
            </button>
            
            <!--&lt;Button&gt;-->
            <!--&lt;AddTokenPrompt&gt;-->
             
            <a
              href="https://blockexplorer.com/address/0x123"
              rel="noopener noreferrer"
              target="_blank"
            >
              <button
                class="h-10 btn btn-theme btn-sm items-center flex justify-center disabled:cursor-not-allowed !rounded-[100px] bg-base-200 text-primary border-none hover:bg-base-300 font-medium gap-3 flex-nowrap"
                id=""
              >
                 
                <img
                  alt="Open"
                  src="/images/openinnew.svg"
                />
                 
                Block Explorer
              </button>
              
              <!--&lt;Button&gt;-->
            </a>
          </div>
        </div>
         
        <div
          class="modal-footer sticky bottom-0 z-[2] bg-white-100 px-6 pb-6 pt-4 s-GZwqZrPaBy6h"
        >
          <button
            class="gap-1.5 h-14 btn btn-theme btn-sm items-center flex justify-center disabled:cursor-not-allowed !rounded-[100px] btn-primary disabled:bg-primary disabled:bg-opacity-60 disabled:text-white w-full font-semibold text-base"
            id=""
          >
             
            LOGOUT
          </button>
          
          <!--&lt;Button&gt;-->
        </div>
         
      </div>
    </div>
    <!--&lt;Modal&gt;-->
    <!--&lt;DisconnectWalletModal&gt;-->
  </div>
</body>
`;
